code/modules/projectiles/projectile/magic.dm

```diff
@@ -451,6 +451,117 @@
 	addtimer(CALLBACK(src, .proc/decay), 15 SECONDS)
 	icon_welded = "welded"
 
+/obj/item/projectile/magic/flying
+	name = "bolt of flying"
+	icon_state = "pulse1_bl"
+
+/obj/item/projectile/magic/flying/on_hit(target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.anti_magic_check())
+			L.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		L.apply_status_effect(STATUS_EFFECT_FLYMAGIC)
+
+/obj/item/projectile/magic/grounding
+	name = "bolt of grounding"
+	icon_state = "pulse1_bl"
+
+/obj/item/projectile/magic/grounding/on_hit(target)
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.anti_magic_check())
+			L.visible_message("<span class='warning'>[src] vanishes on contact with [target]!</span>")
+			return BULLET_ACT_BLOCK
+		if(!(L.movement_type & FLYING))
+			return BULLET_ACT_BLOCK
+		L.visible_message("<span class='danger'>[target] is violently ripped out of the air!</span>")
+		L.setMovetype(L.movement_type | FLYING) //will be used with staff of flying for combos often...
+		L.adjustBruteLoss(L.maxHealth/2)//...because the fly magic will throw them back in the air after getting grounded!
+		if(!L.health)
```

Comment:
Should be <= SOFTCRIT_THRESHOLD or whatever it is now, pretty sure ! will return false if they're under 0.

Reply:
thanks for the heads up